### PR TITLE
[expo-asset] Fix ExpoAsset.web.ts

### PR DESCRIPTION
**Pull request by jamiees2**
# Why

This function doesn't match the declaration in [ExpoAsset.ts](https://github.com/expo/expo/blob/596a7f36f5f34558841c72f73bced1ac38dd28f2/packages/expo-asset/src/ExpoAsset.ts), since the resulting object is `{ default: { downloadAsync: ... }}` instead of `{ downloadAsync: ... }` like ExpoAsset.ts. 

This causes the error `TypeError: (0, _ExpoAsset.downloadAsync) is not a function` when trying to develop with Expo web.

# How

By updating the signature of ExpoAsset.web to match the definition of ExpoAsset

# Test Plan

After running `Asset.loadAsync([require("image.png"])` on Expo web, this previously failed with the aforementioned error, after this patch, it succeeds.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-asset/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 ### 🎉 New features
 
 ### 🐛 Bug fixes
+- Fix `TypeError: (0, _ExpoAsset.downloadAsync) is not a function` when loading assets using Expo Web. ([#28405](https://github.com/expo/expo/pull/28405) by [@jamiees2](https://github.com/jamiees2))
```
Comment (tsapeta) [inline packages/expo-asset/CHANGELOG.md:10]:
```suggestion

- Fix `TypeError: (0, _ExpoAsset.downloadAsync) is not a function` when loading assets using Expo Web. ([#28405](https://github.com/expo/expo/pull/28405) by [@jamiees2](https://github.com/jamiees2))
```

**Review by tsapeta — CHANGES_REQUESTED** [review]
Thanks for fixing this! Could you also run `yarn build` in the package folder to rebuild JS files?

**Comment by jamiees2** [comment]
Done :)

**Review by tsapeta — APPROVED** [review]
Great, thank you!

**Comment by tsapeta** [comment]
Looks like the formatting also needs to be fixed, see the CI job that failed: https://github.com/expo/expo/actions/runs/8808424372/job/24177500392?pr=28405

**Comment by jamiees2** [comment]
ah, that's been fixed now :)

**Review by wschurman — APPROVED** [review]

**Comment by wschurman** [comment]
Looks like the `check-packages` test is still having issues. Running `et cp expo-asset` and then committing the result should fix it. I'll merge once that check is passing. Don't worry about the other checks (Updates e2e) since those need to be run by a expo employee.

**Pull request merged**